ext/CEGUI/IrrlichtBaWTexture.cpp

```diff
@@ -0,0 +1,167 @@
+#include "IrrlichtBaWTexture.hpp"
+
+using namespace irr;
+using namespace ext;
+using namespace cegui;
+
+IrrlichtBaWTexture::IrrlichtBaWTexture(irr::core::smart_refctd_ptr<irr::IrrlichtDevice> _device)
+    : device(_device)
+{
+
+}
+
+IrrlichtBaWTexture::~IrrlichtBaWTexture()
+{
+
+}
+
+const CEGUI::String& IrrlichtBaWTexture::getName() const
+{
+    return cachingName;
+}
+
+const CEGUI::Sizef& IrrlichtBaWTexture::getSize() const
+{
+    bool status = gpuTexture.get();
+
+    if (status)
+    {
+        auto& params = gpuTexture->getCreationParameters().image->getCreationParameters();
+        return { params.extent.width, params.extent.height };
+    }
+    else
+    {
+        assert(status);
+        return { 0, 0 };
+    }
+        
+}
+
+const CEGUI::Sizef& IrrlichtBaWTexture::getOriginalDataSize() const
+{
+    bool status = gpuTexture.get();
+
+    if (status)
+    {
+        auto& params = gpuTexture->getCreationParameters().image->getCreationParameters();
+        return { bufferRowLength ? bufferRowLength : params.extent.width, params.extent.height };
+    }
+    else 
+    {
+        assert(status);
+        return { 0, 0 };
+    }
+}
+
+const CEGUI::Vector2f& IrrlichtBaWTexture::getTexelScaling() const
+{
+    bool status = gpuTexture.get();
+
+    if (status)
+    {
+        auto& params = gpuTexture->getCreationParameters().image->getCreationParameters();
+        auto width = bufferRowLength ? bufferRowLength : params.extent.width;
+
+        return
+        {
+            static_cast<float>(width) == 0.f ? 0.f : 1.f / static_cast<float>(width),
+            static_cast<float>(params.extent.height) == 0.f ? 0.f : 1.f / static_cast<float>(params.extent.height)
+        };
+    }
+    else
+    {
+        assert(status);
+        return { 0, 0 };
+    }
+}
+
+void IrrlichtBaWTexture::loadFromFile(const CEGUI::String& filename, const CEGUI::String& resourceGroup)
+{
+    auto path = resourceGroup + filename;
+    auto assetManager = device->getAssetManager();
+
+    auto params = asset::IAssetLoader::SAssetLoadParams();
+    auto bundle = assetManager->getAsset(path, params);
```

Comment:
you could possibly load with an override that overrides the cache searching so you get a separate asset cache just for CEGUI (you'd also need to override the GPU object creator slightly for `getGPUObjectsFromAssets`)

CC: @Crisspl